econpizza/config.py

```diff
@@ -0,0 +1,80 @@
+"""Configuration object"""
+
+import os
+import jax
+
+class EconPizzaConfig(dict):
+    def __init__(self, *args, **kwargs):
+        super(EconPizzaConfig, self).__init__(*args, **kwargs)
+        self._enable_persistent_cache = False
+        self._econpizza_cache_folder = "__econpizza_cache__"
+        self._jax_cache_folder = "__jax_cache__"
+
+    @property
+    def enable_persistent_cache(self):
+        return self._enable_persistent_cache
+
+    @enable_persistent_cache.setter
+    def enable_persistent_cache(self, value):
+        self._enable_persistent_cache = value
+        self.setup_persistent_cache()
+
```

Comment:
@gboehl , for the automatic setup of cache is this something you see fit as a solution?

Reply:
I think just setting `ep.config.update("enable_persistent_cache", True)` should be enough. This way there is a single interface to the config, that is, `ep.config`. 

You can extend the update function such that the rest happens automatically.

Reply:
This is how it works now, yes. The user only calls
`ep.config.update("enable_persistent_cache", True)`

Then, we go into the `update` function, which calls each of the public properties' setters. Then, we enter the setter of `enable_persistent_cache` and call `setup_persistent_cache()`.

I added a test file as well, all cases that I can think of are covered.

Reply:
Either I don't understand what you mean, or we are talking about different things. I think the user should just set  `ep.config.update("enable_persistent_cache", True)` and that's it. I don't think that it should be necessary to call any other function. The `update` function can be written such that it does anything else automatically, but right now it is just setting the `enable_persistent_cache` bool. Where is the point where we diverge?

Reply:
The `update` function works for any other config option that is exposed (`ep.config.update("enable_x", True)`).

 I agree its a little overkill, but on the other hand, if there are more config options in the future the API will stay the same. The `setup_persistent_cache` needs to be called only when `enable_persistent_cache` is called and that's why its invoked in the setter, not in the update function.

Reply:
I still don't understand, but never mind. Better focus on caching. I'll take care of that myself later.